File: app/src/main/java/era/apps/happinessjar/models/chat/model/ChatViewModel.java
package era.apps.happinessjar.models.chat.model;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;




public class ChatViewModel extends AndroidViewModel {
    private final ChatMessageRepository repository;

    public ChatViewModel(@NonNull Application application) {
        super(application);
        String id = application.getSharedPreferences("info", 0).getString("chatId", "");
        repository = ChatMessageRepository.getInstance(id);
        allChatMessage = repository.getChatMessage();
    }

    private final LiveData<Conversation> allChatMessage;

    public LiveData<Conversation> getAllChatMessage() {
        return allChatMessage;
    }

    public void setOnMessageSent(OnMessageSent onMessageSent) {
        repository.setOnMessageSent(onMessageSent);
    }

    public void sendMessage(Conversation conversation) {
        repository.sendMessage(conversation);
    }

}

File: app/src/main/java/era/apps/happinessjar/ui/viewmodel/ChatViewModel.kt
package era.apps.happinessjar.ui.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import era.apps.happinessjar.data.models.Conversation
import era.apps.happinessjar.data.repository.ChatMessageRepository
import era.apps.happinessjar.util.callback.OnMessageSent


class ChatViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: ChatMessageRepository
    private val allChatMessage: LiveData<Conversation>

    fun getAllChatMessage():LiveData<Conversation>{
        return allChatMessage
    }


    fun sendMessage(conversation: Conversation) {
        repository.sendMessage(conversation)
    }

    fun update(conversation: Conversation) {
        repository.updateMessage(conversation)
    }

    init {
        val id = application.getSharedPreferences("info", 0).getString("chatId", "")
        repository = ChatMessageRepository.getInstance(id!!)!!
        allChatMessage = repository.chatMessage
    }
}